electrum/tests/test_blockchain.py

```diff
@@ -339,3 +331,18 @@ def test_doing_multiple_swaps_after_single_new_header(self):
 
         for b in (chain_u, chain_l, chain_z):
             self.assertTrue(all([b.can_connect(b.read_header(i), False) for i in range(b.height())]))
+
+    def test_mainnet_get_chainwork(self):
+        constants.set_mainnet()
+        blockchain.blockchains[constants.net.GENESIS] = chain_u = Blockchain(
+            config=self.config, forkpoint=0, parent=None,
+            forkpoint_hash=constants.net.GENESIS, prev_hash=None)
+        open(chain_u.path(), 'w+').close()
+
+        # Try a variety of checkpoint heights relative to the chunk boundary
+        for height_offset in range(2016):
+            constants.net.CHECKPOINTS['height']+=1
+
+            chain_u.get_chainwork(constants.net.max_checkpoint())
+            with self.assertRaises(MissingHeader):
+                chain_u.get_chainwork(constants.net.max_checkpoint() + 4032)
```

Comment:
What is this supposed to test? (the ~10 lines above)
Just whether get_chainwork raises or not?
Is it intentional that you are changing `constants.net.CHECKPOINTS['height']` ? and that `height_offset` is not used?

Reply:
Yeah, basically this test is supposed to ensure that `get_chainwork`'s assert behavior is correct regardless of what the checkpoint height MOD the retargeting period is equal to.  Early drafts of this PR were broken for a subset of possible moduli, so I figured an exhaustive test of that space would avoid any accidental future regressions in that area.  I think height_offset's value isn't used for anything, I just needed a variable name for the for loop, so I picked one that was hopefully self-documenting of what the range was.